### ISSUE TITLE
Make portals4.h c++ compatible by default

### ISSUE THREAD

**Issue by EloWork** [Portals4/portals4#86]
Trying to compile c++ with portals in it will result in an undefined reference error during link-time because the symbol names are different in C than C++. One can fix that by wrapping the `#include` in an `extern C` statement. However, it takes some digging to find that out. Usually, libraries take care of that themselves by wrapping their headers with 
```
#ifdef __cplusplus
extern "C" {
#endif

#ifdef __cplusplus
}
#endif
```

**Comment by tkordenbrock** [comment]
Thanks for posting an issue.  I will create a PR for this shortly.

**Comment by tkordenbrock** [comment]
Fixed in #87
